### PR TITLE
Add a language selector to the Mismatch Finder layout

### PR DESCRIPTION
**Pull request by chukarave**
This allows changing the interface language from the UI.
(rather a large PR but i couldn't think of a way to split it that made sense)

Bug: [T328149](https://phabricator.wikimedia.org/T328149)

### CODE REVIEW

**Comment by chukarave** [comment]
I get the above errors locally as well, although the SVGs are displayed in the UI. Might be a path issue

File: resources/js/Components/LanguageSelectorOptionsMenu.vue [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			if ( child === null ) {
+				return;
+			}
+
+			const container = ( this.$refs.container as Element ).getBoundingClientRect(),
+				item = child.getBoundingClientRect(),
+				above = Math.floor( item.top ) < container.top,
+				below = Math.ceil( item.bottom ) > container.bottom;
+
+			if ( !( above || below ) ) {
```
[the remark this answers is not in the document]
Comment (itamargiv) [reply resources/js/Components/LanguageSelectorOptionsMenu.vue:72]: Good call! thanks, I will change it (even though this is a direct copy paste from QB, but meh, It's going to be migrated soon anyhow, so why not improve it a little)

File: resources/js/Components/LanguageSelectorInput.vue [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			<div class="languageSelector__input__search-icon">
+				<img :src="searchUrl" alt="">
+			</div>
+			<input
+				ref="input"
+				type="text"
+				class="languageSelector__input"
+				:value="value"
+				:placeholder="placeholder"
+				@input="$emit( 'input', $event.target.value )"
```
[the remark this answers is not in the document]
Comment (itamargiv) [reply resources/js/Components/LanguageSelectorInput.vue:13]: You mean to create another handler and move this logic to it like so?

```js
@input="onInput"
//...
onInput(event){
  this.$emit('input', event.target.value)
}
```

[the remark this answers is not in the document]
Comment (hasanakg) [reply resources/js/Components/LanguageSelectorInput.vue:13]: Exactly!

[the remark this answers is not in the document]
Comment (itamargiv) [reply resources/js/Components/LanguageSelectorInput.vue:13]: :+1:

**Pull request merged**